lockup/README.md

```diff
@@ -1,43 +1,64 @@
 # Lockup / Vesting contract
 
-# PLEASE WAIT UNTIL [ISSUE 141](https://github.com/near/core-contracts/issues/141) IS RESOLVED BEFORE DEPLOYING THE CONTRACT
-
 ## Overview
 
-This contract acts as an escrow that locks and holds an owner's tokens for a lockup period. A lockup period starts
-from the moment transfers are enabled by voting and lasts for the specified duration.
+This contract acts as an escrow that locks and holds an owner's tokens for a lockup period. 
+The contract contains of lockup and vesting processes that go simultaneously.
+
+A lockup period starts from the specified timestamp (or, if not specified, from moment when transfers are enabled by voting) 
+and lasts for the specified duration.
 It's also possible to lock tokens until the absolute timestamp. In this case the tokens will be available after the
 lockup period or after the timestamp, whichever is later.
 
-If transfers are not enabled yet, the contract keeps the account ID of the transfer poll contract.
-When the transfer poll is resolved, it returns the timestamp when it was resolved and it's used as the beginning of the
-lockup period.
+Vesting is an additional mechanism. It also locks the tokens, and it allows to configure 2 more options:
+1. Ability of terminating the contract from the Foundation.
+2. Cliff: the period when all tokens remain locked. After the cliff, there is a linear unlock till the end of the contract.
```

Comment:
Maybe something like this:

```suggestion
2. Cliff vesting period. All tokens remain non-vested before the cliff checkpoint is reached; on the cliff checkpoint, the accumulated tokens are vested in bulk; after the cliff, tokens are linearly unlocked till the vesting reaches its end timestamp.
```

or, just:

```suggestion
2. Cliff vesting period.
```

(the details are provided below)